sharedLibNetTests/TestMeterMonitor.cs

```diff
@@ -14,19 +19,55 @@ public class TestMeterMonitor
         /// <summary>
         /// just for local test
         /// </summary>
-        [Fact]
-        public async void TestGetAll()
+        [Fact(Skip = "just for local test")]
+        public async Task TestGetAll()
         {
             var result = LoggerHelper.CreateLogger("Dei Mudder sein Service", null, null);
             var logger = result.logger;
-            MeterMonitorHelper mmrHelper = new MeterMonitorHelper(logger,new System.Net.Http.HttpClient() { BaseAddress = new Uri("https://hfapi-stage.azure-api.net/metermonitor/"), Timeout = TimeSpan.FromMinutes(10) });
+            MeterMonitorHelper mmrHelper = new MeterMonitorHelper(logger, new System.Net.Http.HttpClient() { BaseAddress = new Uri("https://hfapi-stage.azure-api.net/metermonitor/"), Timeout = TimeSpan.FromMinutes(10) });
             string token = "";
             string apiKey = "";
-            var top1000Results = await mmrHelper.GetMeterMonitors(token, apiKey, new EshDataExchangeFormats.lookup.BOBackendId("HOCHFREQUENZ_BASIC_AUTH"),limit: 1000);
+            var top1000Results = await mmrHelper.GetMeterMonitors(token, apiKey, new EshDataExchangeFormats.lookup.BOBackendId("HOCHFREQUENZ_BASIC_AUTH"), limit: 1000);
             //Assert.Equal(1000, top1000Results.Count);
-            var allResults = await mmrHelper.GetMeterMonitors(token, apiKey, new EshDataExchangeFormats.lookup.BOBackendId("HOCHFREQUENZ_BASIC_AUTH"),withError:false);
+            var allResults = await mmrHelper.GetMeterMonitors(token, apiKey, new EshDataExchangeFormats.lookup.BOBackendId("HOCHFREQUENZ_BASIC_AUTH"), withError: false);
             //Assert.True(allResults.Count> 70000);
 
         }
+
+        [Fact]
+        public async Task TestMeterMonitorHelperMocker()
+        {
+            string json = "";
+            using (StreamReader r = new StreamReader("test_data\\getall-16121856537571.json"))
+            {
+                json = r.ReadToEnd();
+            }
+            var handlerMock = new Mock<HttpMessageHandler>();
+            var response = new HttpResponseMessage
+            {
+                StatusCode = HttpStatusCode.OK,
+                Content = new StringContent(json),
+            };
+
+            handlerMock.Protected()
+               .Setup<Task<HttpResponseMessage>>(
+                  "SendAsync",
+                  ItExpr.IsAny<HttpRequestMessage>(),
```

Comment:
an dieser stelle wäre eine assertion interessant, was _genau_ eigentlich geschickt wird.
Dieser Helper ist ja dazu da, den HTTP-Aufruf an den lookup-service zu kapseln, d.h. der Test sollte überprüfen, wie genau der HTTP-Aufruf aussieht. z.B. dass die URL auf /meter_pod oder so endet, dass bestimmmte query parameter gesetzt sind.

Reply:
Aber hier sind wir in SharedLibNet und `/meter_pod` und query parameters sind in LookupService?!

Reply:
ich dachte der helper macht den http request?

Reply:
Jup macht ein http request aber nach LookupService. die SAP-HTTP request liegt in LookupService. Kann ich da auch eine ähnliches test bauen nach SAP?

Reply:
der test sollte sicherstellen, dass die richtige URL im lookup service aufgerufen wird.